packages/apputils/src/thememanager.ts

```diff
@@ -60,20 +60,77 @@ export class ThemeManager implements IThemeManager {
     return this._current;
   }
 
+  /**
+   * Get the name of the preferred light theme.
+   */
+  get preferredLightTheme(): string {
+    return this._settings.composite['preferred-light-theme'] as string;
+  }
+
+  /**
+   * Get the name of the preferred dark theme.
+   */
+  get preferredDarkTheme(): string {
+    return this._settings.composite['preferred-dark-theme'] as string;
+  }
+
+  /**
+   * Get the name of the preferred theme
+   * When `adaptive-theme` is disabled, get current theme;
+   * Else, depending on the system settings, get preferred light or dark theme.
+   */
+  get preferredTheme(): string | null {
+    if (!this.isToggledAdaptiveTheme()) {
+      return this.theme;
+    }
+    if (this.isOSColorSchemeDark()) {
+      return this.preferredDarkTheme;
+    }
+    return this.preferredLightTheme;
+  }
+
   /**
    * The names of the registered themes.
    */
   get themes(): ReadonlyArray<string> {
     return Object.keys(this._themes);
   }
 
+  /**
+   * Get the names of the light themes.
+   */
+  get lightThemes(): ReadonlyArray<string> {
+    return Object.entries(this._themes)
+      .filter(([_, theme]) => theme.isLight)
+      .map(([name, _]) => name);
+  }
+
+  /**
+   * Get the names of the dark themes.
+   */
+  get darkThemes(): ReadonlyArray<string> {
+    return Object.entries(this._themes)
+      .filter(([_, theme]) => !theme.isLight)
+      .map(([name, _]) => name);
+  }
+
   /**
    * A signal fired when the application theme changes.
    */
   get themeChanged(): ISignal<this, IChangedArgs<string, string | null>> {
     return this._themeChanged;
   }
 
+  /**
+   * Test if the preferred OS color scheme is dark
+   */
+  isOSColorSchemeDark(): boolean {
```

Comment:
Maybe the following would be more coherent since `system` seems to be used elsewhere:

```suggestion
  isSystemColorSchemeDark(): boolean {
```